datadog/internal/fwutils/fw_enrich_schema.go

```diff
@@ -81,9 +81,62 @@ func buildEnrichedSchemaDescription(rv reflect.Value) {
 					}
 					validValuesMsg += fmt.Sprintf("%s`%v`", sep, v.Index(i).Interface())
 				}
-				curentDesc = fmt.Sprintf("%s Valid values are %s.", curentDesc, validValuesMsg)
+				currentDesc = fmt.Sprintf("%s Valid values are %s.", currentDesc, validValuesMsg)
 				break
 			}
+			if strings.HasPrefix(validators.Index(i).Elem().Type().Name(), "oneOfValidator") {
+				allowedValues := validators.Index(i).Elem().FieldByName("values")
+				validValuesMsg := ""
+				sep := ""
+				for i := 0; i < allowedValues.Len(); i++ {
+					if len(validValuesMsg) > 0 {
+						sep = ", "
+					}
+					// Index(i).Field(1) is the value of the types.String
+					// If we would use "only" Index(i) we would have { 2 <VALUE> }
+					validValuesMsg += fmt.Sprintf("%s`%v`", sep, allowedValues.Index(i).Field(1))
+				}
+				currentDesc = fmt.Sprintf("%s. Valid values are %s.", currentDesc, validValuesMsg)
+				break
+			}
+			if strings.HasPrefix(validators.Index(i).Elem().Type().Name(), "regexMatchesValidator") {
+				validationMessage := validators.Index(i).Elem().FieldByName("message").String()
+				currentDesc = fmt.Sprintf("%s %s", currentDesc, validationMessage)
+				break
+			}
+			if strings.HasPrefix(validators.Index(i).Elem().Type().Name(), "validEntityYAMLValidator") {
+				currentDesc = fmt.Sprintf("%s entity must be a valid entity YAML/JSON structure.", currentDesc)
```

Comment:
I think we could use the `Description` method from the validator struct here rather than re-writing this description.
This is also true for the other validators.
This way we can probably make all the `if`s in here behave in the same way and refactor quite a bit.